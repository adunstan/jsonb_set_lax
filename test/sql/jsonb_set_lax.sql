


\pset null NULL

-- pass though non nulls to jsonb_set
select jsonb_set_lax('{"a":1,"b":2}','{b}','5') ;
select jsonb_set_lax('{"a":1,"b":2}','{d}','6', true) ;

-- using the default treatment
select jsonb_set_lax('{"a":1,"b":2}','{b}',null);
select jsonb_set_lax('{"a":1,"b":2}','{d}',null,true);

-- errors
select jsonb_set_lax('{"a":1,"b":2}', '{b}', null, true, null);
select jsonb_set_lax('{"a":1,"b":2}', '{b}', null, true, 'no_such_treatment');

-- explicit treatments
select jsonb_set_lax('{"a":1,"b":2}', '{b}', null, null_value_treatment => 'raise_exception') as raise_exception;
select jsonb_set_lax('{"a":1,"b":2}', '{b}', null, null_value_treatment => 'return_target') as return_target;
select jsonb_set_lax('{"a":1,"b":2}', '{b}', null, null_value_treatment => 'delete_key') as delete_key;
select jsonb_set_lax('{"a":1,"b":2}', '{b}', null, null_value_treatment => 'use_json_null') as use_json_null;
