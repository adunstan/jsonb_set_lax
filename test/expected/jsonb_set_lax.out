\pset null NULL
select jsonb_set_lax('{"a":1,"b":2}','{b}',null);
    jsonb_set_lax    
---------------------
 {"a": 1, "b": null}
(1 row)

select jsonb_set_lax('{"a":1,"b":2}','{d}',null,true);
        jsonb_set_lax        
-----------------------------
 {"a": 1, "b": 2, "d": null}
(1 row)

select jsonb_set_lax('{"a":1,"b":2}', '{b}', null, true, null);
ERROR:  need delete_key, return_target, use_json_null, or raise_exception
select jsonb_set_lax('{"a":1,"b":2}', '{b}', null, null_value_treatment := 'raise_exception');
ERROR:  null jsonb value
select jsonb_set_lax('{"a":1,"b":2}', '{b}', null, null_value_treatment := 'return_target');
  jsonb_set_lax   
------------------
 {"a": 1, "b": 2}
(1 row)

select jsonb_set_lax('{"a":1,"b":2}', '{b}', null, null_value_treatment := 'delete_key');
 jsonb_set_lax 
---------------
 {"a": 1}
(1 row)

